Userland/Libraries/LibWeb/CSS/StyleValue.cpp

```diff
@@ -1432,6 +1439,26 @@ bool ImageStyleValue::equals(StyleValue const& other) const
     return m_url == other.as_image().m_url;
 }
 
+Optional<int> ImageStyleValue::natural_width() const
+{
+    if (m_bitmap)
+        return m_bitmap->width();
+    return {};
+}
+
+Optional<int> ImageStyleValue::natural_height() const
+{
+    if (m_bitmap)
+        return m_bitmap->height();
+    return {};
+}
+
+void ImageStyleValue::paint(PaintContext& context, Gfx::IntRect const& dest_rect) const
+{
+    if (m_bitmap)
+        context.painter().draw_scaled_bitmap(dest_rect, *m_bitmap, m_bitmap->rect(), 1.0f, Gfx::Painter::ScalingMode::BilinearBlend);
```

Comment:
I know I'm super-duper late on this, but `image-rendering` is supposed to apply to all images in an element, so that would require passing in the ScalingMode here. That's not a huge change though probably.

> When specified on an element, it applies to all images given in properties for the element, such as background images, list-style images, or the content of replaced elements when they represent an image that must be scaled.

https://www.w3.org/TR/2020/CRD-css-images-3-20201217/#propdef-image-rendering

Reply:
Sounds easy enough, I'll sort it out in a little bit :)